doc/query.md

```diff
@@ -14,8 +14,15 @@ This query engine uses indexes explicitly defined over the fields in the documen
 indexes can be created for use in different queries, the same field may end up indexed in
 more than one index.
 
-Querying is carried out by supplying a query in the form of a dictionary which describes the
-query.
+Query offers a powerful way to find documents within your datastore. There are a couple of restrictions on field names you need to be aware of before using query:
+
+- A dollar sign (`$`) cannot be the first character of any field name.  This is because, when querying, a dollar sign tells the query engine to handle the object as a query operator and not a field.
+- A field with a name that contains a period (`.`) cannot be indexed.  This is because using dot notation tells the query engine to index a sub-object when creating an index.
```

Comment:
Does this mean, however, that a dot-notated field can be searched for in the post-hoc matcher? That is, given a set of docs like
```
{
  name : mike
  age : 34
  pet.species : cat
  pet.name : bob
}
```
and an index a field "name", can I make the query `{selector : {name : mike, pet.species = cat}}`?